keylime-agent/Cargo.toml

```diff
@@ -1,47 +1,47 @@
 [package]
-authors = ["Keylime Authors"]
-edition = "2021"
 name = "keylime_agent"
-version = "0.2.3"
-license = "Apache-2.0"
 description = "Rust agent for Keylime"
-repository = "https://github.com/keylime/rust-keylime"
+authors.workspace = true
+edition.workspace = true
+license.workspace = true
+repository.workspace = true
+version.workspace = true
 
 [dependencies]
-actix-web =  { version = "4", default-features = false, features = ["macros", "openssl"] }
-base64 = "0.21"
-cfg-if = "1"
-clap = { version = "4.3", features = ["derive"] }
-compress-tools = "0.12"
-config = { version = "0.13", default-features = false, features = ["toml"] }
-futures = "0.3.6"
-glob = "0.3"
-hex = "0.4"
-keylime = { path = "../keylime" }
-libc = "0.2.43"
-log = "0.4"
-openssl = "0.10.15"
-picky-asn1-der = "0.3.1"
-picky-asn1-x509 = "0.6.1"
-pretty_env_logger = "0.4"
-reqwest = {version = "0.11", default-features = false, features = ["json"]}
-serde = "1.0.80"
-serde_derive = "1.0.80"
-serde_json = { version = "1.0", features = ["raw_value"] }
-static_assertions = "1"
-tempfile = "3.4.0"
-tokio = {version = "1.24", features = ["rt", "sync", "macros"]}
-tss-esapi = {version = "7.2.0", features = ["generate-bindings"]}
-thiserror = "1.0"
-uuid = {version = "1.3", features = ["v4"]}
+actix-web.workspace = true
+base64.workspace = true
+cfg-if.workspace = true
+clap.workspace = true
+compress-tools.workspace = true
+config.workspace = true
+futures.workspace = true
+glob.workspace = true
+hex.workspace = true
+keylime.workspace = true
+libc.workspace = true
+log.workspace = true
+openssl.workspace = true
+picky-asn1-der.workspace = true
+picky-asn1-x509.workspace = true
+pretty_env_logger.workspace = true
+reqwest.workspace = true
+serde.workspace = true
+serde_derive.workspace = true
+serde_json.workspace = true
+static_assertions.workspace = true
+tempfile.workspace = true
+tokio.workspace = true
+tss-esapi.workspace = true
+thiserror.workspace = true
+uuid.workspace = true
 zmq = {version = "0.9.2", optional = true}
 # wiremock was moved to be a regular dependency because optional
 # dev-dependencies are not supported
 # see: https://github.com/rust-lang/cargo/issues/1596
 wiremock = {version = "0.5", optional = true}
```

Comment:
Any reason why zmq and wiremock are out of the inheritance?

Reply:
I think because they are used only in `keylime-agent`, not shared with the other parts (`keylime` and `keylime-ima-emulator`)

Reply:
The reason behind this is that the top-level workspace.dependencies can't include optional dependencies, as described at https://doc.rust-lang.org/cargo/reference/workspaces.html#the-dependencies-table

Reply:
@ueno I can be confused here, so bare with me. But IIUC this is valid:

`zmq = {workspace = ture, optional = true}`

See for example https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#inheriting-a-dependency-from-a-workspace

You are right that we cannot do this in the top level, in `[workspace.dependencies]`, where all the dependencies are listed, but I am not sure how relevant is that.  For example, if a consequence of doing this is that during vendoring the optional crates will always be included.